### PR TITLE
Add search functionality to user admin index.

### PR DESCRIPTION
**Pull request by DFurnes**
### What's this PR do?

This pull request adds support for searching for a user in "Roguestar" admin.

### How should this be reviewed?

I extracted the logic we use to recognize if something is a phone number or email in ae31642 & ce86f54. Using those helpers, I created a `search` query scope that infers what term is given & queries the appropriate field in ba0acb9.

Finally, I added the search form & controller support (as a simple `?search=` query on the index) in ed14d86.

### Any background context you want to provide?

I hadn't moved the "search" interface from Aurora into this app in [#176850876](https://www.pivotaltracker.com/story/show/176850876). This is an important use-case for Aurora and Rogue, and something we'll need to support before we can turn off the old admin interfaces!

### Relevant tickets

References [Pivotal #176911792](https://www.pivotaltracker.com/story/show/176911792).

### Checklist

- [ ] This PR has been added to the relevant Pivotal card.
- [ ] Documentation added for new features/changed endpoints.
- [ ] Added appropriate feature/unit tests.
- [ ] If new attributes were added to users, there is a corresponding PR to surface these in Aurora.
- [ ] If new attributes were added to users, then the data team already knows about these changes.

### CODE REVIEW

File: app/helpers.php [hunk 1]
```diff
@@ -213,6 +213,58 @@ function format_birthdate($birthdate)
     return format_date($birthdate, 'Y-m-d');
 }
 
+/**
+ * Is the given value an email address?
```
Comment (mendelB) [inline app/helpers.php:217]:
```suggestion
 * Is the given value a phone number?
```

👾

File: app/helpers.php [hunk 2]
```diff
@@ -213,6 +213,58 @@ function format_birthdate($birthdate)
...
+        );
+
+        return !empty($valid) && empty($repeat);
+    } catch (\libphonenumber\NumberParseException $e) {
+        return false;
+    }
+}
+
+/**
+ * Is the given value a phone number?
```
Comment (mendelB) [inline app/helpers.php:254]:
```suggestion
 * Is the given value an email address?
```

**Review by mendelB — APPROVED** [review]
💥

File: app/helpers.php (hunk 1 above)
Comment (DFurnes) [reply app/helpers.php:217]: Oh my god, how did I get them swapped?! 😂

Comment (mendelB) [reply app/helpers.php]: Email is the new SMS! (so Anthony claims 🙃)

**Pull request merged**